src/mixins/render-vega-lite/Definitions/Transform/Transforms/CrossSectionTerrainDefinitionObject.js

```diff
@@ -0,0 +1,320 @@
+import assert from "assert"
+import PropertiesDefinitionInterface from "../../PropertiesDefinitionInterface"
+import { PositionChannelDescriptor } from "../../../PropDescriptor/CommonChannelDescriptors"
+
+// jsdoc imports only
+/* eslint-disable no-unused-vars */
+import VegaPropertyOutputState from "../../../VegaPropertyOutputState"
+/* eslint-enable no-unused-vars */
+
+export default class CrossSectionTerrainDefinitionObject extends PropertiesDefinitionInterface {
+  // eslint-disable-next-line no-undef
+  static key = "cross_section1d"
+
+  /**
+   * @param {Object} definition_object
+   * @param {ParentInfo} parent_info
+   */
+  constructor(definition_object, parent_info) {
+    super(definition_object, parent_info)
+    assert(
+      Object.hasOwn(definition_object, CrossSectionTerrainDefinitionObject.key)
+    )
+
+    const key = CrossSectionTerrainDefinitionObject.key
+
+    const obj = definition_object[key]
+    if (typeof obj !== "object") {
+      throw new Error(
+        `Invalid 1D cross-section transform definition. The '${key}' property must be an object`
+      )
+    }
+
+    /**
+     * @type {string}
+     */
+    this.x_ = ""
+    if (typeof obj.x !== "string") {
+      throw new Error(
+        `Invalid '${key}' transform definition. The 'x' property must exist and be a string`
+      )
+    }
+    this.x_ = obj.x
+
+    /**
+     * @type {string}
+     */
+    this.y_ = ""
+    if (typeof obj.x !== "string") {
+      throw new Error(
+        `Invalid '${key}' transform definition. The 'y' property must exist and be a string`
+      )
+    }
+    this.y_ = obj.y
+
+    /**
+     * @type {string}
+     */
+    this.z_ = ""
+    if (typeof obj.z !== "string") {
+      throw new Error(
+        `Invalid '${key}' transform definition. The 'z' property must exist and be a string`
+      )
+    }
+    this.z_ = obj.z
+
+    /**
+     * @type {Number[]}
+     */
+    this.cross_section_line_ = []
+    if (
+      !Array.isArray(obj.crossSectionLine) ||
```

Comment:
Sheesh, could be a good opportunity to make this more readable and make a `validateCrossSectionLine` method or something. Not required

```
!Array.isArray(obj.crossSectionLine[1]) ||
      obj.crossSectionLine[1].length !== 2 ||
      !obj.crossSectionLine[1].every(p =>
        ["number", "string"].includes(typeof p)
      )```  
could def be pulled out into a function and called for each set of coords in the `crossSectionLine`